runtime/vm/resolvefield.cpp

```diff
@@ -578,6 +581,103 @@ addHiddenInstanceField(J9JavaVM *vm, const char *className, const char *fieldNam
 	return 0;
 }
 
+/**
+ * Report a hot field if the JIT has determined that the field has met appropriate thresholds to be determined a hot field. 
+ * Valid if dynamicBreadthFirstScanOrdering is enabled.
+ *
+ * @param vm[in] pointer to the J9JavaVM
+ * @param reducedCpuUtil normalized cpu utilization of the hot field for the method being compiled
+ * @param clazz pointer to the class where a hot field should be added
+ * @param fieldOffset value of the field offset that should be added as a hot field for the given class
+ * @param reducedFrequency normalized block frequency of the hot field for the method being compiled
+ */
+void
+reportHotField(J9JavaVM *javaVM, int32_t reducedCpuUtil, J9Class* clazz, U_8 fieldOffset,  U_32 reducedFrequency)
+{
+	/* exit if pools are null */
+	if (clazz->classLoader->hotFieldPool == NULL || javaVM->hotFieldClassInfoPool == NULL || javaVM->hotFieldClassInfoPoolMutex == NULL) {
+		return;
+	}
+
+	/* check if hotFieldsInfo exists already */
+	if(NULL == clazz->hotFieldsInfo) { 
+		omrthread_monitor_enter(javaVM->hotFieldClassInfoPoolMutex);
+		if(NULL == clazz->hotFieldsInfo) {
+         /* initialize new hotFieldsInfo pool element as it does not exist already */
```

Comment:
indentation for this comment and several other lines later below in this method does not now seem right